docs/master-overview.md

```diff
@@ -0,0 +1,39 @@
+# Master PCB
+
+The Master PCB is responsible for calculating and publishing the AS state that the car should be in and performing all the necessary actions required to achieve it. The state is correctly identifiable by the following flowchart, as required by rule **T14.9.2**: "*The status of the AS must be determined according to the flowchart in Figure 17.*
+
+![ASSFlow](./assets/master-overview/ASS-flow.png)
+
+Although the above diagram can determine the state, **it is only meant for externally determining the ASState**, it is **not** how the state is calculated.
+
+## State Machine
+The PCB reads values from all the hardware connected to it, as well as CAN, to then calculate the state it needs to be in (change may or not be necessary), based on the state it currently is in. The "default" state is AS_OFF, and then each state onward is calculated by the following state machine:
+
+![AS State Machine](./assets/master-overview/AS-state-machine.jpg)
+
+Inside we can see all the requirements to transition from one state to another. The Initial Checkup Sequence is crucial for the transition to AS_Ready and is as follows:
+
+![ASB Initial Checkup Sequence](./assets/master-overview/ASB%20Initial%20Checkup%20Flowchart.jpg)
+
+In AS_Ready and AS_Driving, we need to continuously monitor the car in order to verify no system failures or emergencies in general arise:
+
+![ASB Continuous Monitoring Sequence](./assets/master-overview/ASB%20Continuous%20Monitoring%20Flowchart.jpg)
+
+## Main Loop Sequence
+### Set-up
+Before calculating the state, we need to define some CAN callbacks. They are used to receive RES signals, brake pressure, TS state, wheel information, mission information, emergencies, and timestamps (the computing unit, inversor, and steering pcb all need to send "alive" signals at a fixed rate, to confirm they operational). Those  callbacks are asynchronous, which means that they will shortly interrupt the main loop when data from a topic the master node is subscribed to is received to update the relevant variables. We also need to define the operation modes of all the pins that we will be sending and receiving information from.
+### Loop
+During the main loop, we first read all the values from the necessary components (Sensors, AMI, Switches, SDC Logic, EBS, ASSI). After this we calculate and update the state through a sequence of monitoring sequences, updating components' variables as depicted in the following diagram. After this, state and other variables are sent out to the rest of the system.
```

Comment:
AMI and ASSI are not reads

Reply:
fixed, but from where does the Mission read come from then?